go/client/cmd_login.go

```diff
@@ -134,9 +158,26 @@ func (c *CmdLogin) ParseArgv(ctx *cli.Context) error {
 		}
 	}
 	c.doUserSwitch = ctx.Bool("switch")
+
+	c.PaperKey = c.getOption(ctx, "paperkey")
+	c.DeviceName = c.getOption(ctx, "devicename")
+
 	return nil
 }
 
+func (c *CmdLogin) getOption(ctx *cli.Context, s string) string {
+	v := ctx.String(s)
+	if len(v) > 0 {
+		return v
+	}
+	envVarName := fmt.Sprintf("KEYBASE_%s", strings.ToUpper(strings.Replace(s, "-", "_", -1)))
```

Comment:
you know what, let's kill the environment variable idea here, since no other cmd_foo is doing it. Sorry about that. So either via the flag, or via stdin....

Reply:
I like it though! Lots of CI stuff is done using environment variables. I'd totally keep it.